### PR TITLE
DOC: Fix HDF5 complevel and complib formatting

### LINKED ISSUE #31052: User guide HDF5 complevel and complib formatting strange
Formatting for HDF5 `complevel` and `complib` is a bit weird on both the old docs and new docs.

https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#compression
https://pandas.io/docs/user_guide/io.html#compression

### PR DESCRIPTION
**Pull request by kylekeppler**
Fixes formatting of these params.

- [x] closes #31052
- [ ] tests added / passed (N/A)
- [ ] passes `black pandas` (N/A)
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff` (N/A)
- [ ] whatsnew entry (N/A)

N/A as this PR only updates an rst file for the docs.

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -4220,46 +4220,49 @@ Compression
...
-                  somewhat slower than the previous ones, but
-                  achieving better compression ratios.
-                - `blosc:zstd <https://facebook.github.io/zstd/>`_: An
-                  extremely well balanced codec; it provides the best
-                  compression ratios among the others above, and at
-                  reasonably fast speed.
-
-             If ``complib`` is defined as something other than the
-             listed libraries a ``ValueError`` exception is issued.
+complevel : specifies if and how hard data is to be compressed.
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/io.rst]: I would keep those names as code-formatted (double backticks), as it's about keywords in the code

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR!

I am not fully sure this will actually change a lot in the final formatting (it's still formatted as a rst definition list, http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks). 
I was thinking to just make it 2 bullet points? (the complevel and complib)

File: doc/source/user_guide/io.rst (hunk 1 above)
Comment (kylekeppler) [reply doc/source/user_guide/io.rst]: I tried to mimic the formatting used for [other parts of the file](https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#basic). Do you prefer the double-backtick formatting?

Comment (kylekeppler) [reply doc/source/user_guide/io.rst]: Changing to bullets and re-adding backticks.

**Comment by jreback** [comment]
can you post a screen shot of the rendered

**Review by jorisvandenbossche — APPROVED** [review]
Looks good, thanks!

**Comment by jorisvandenbossche** [comment]
> can you post a screen shot of the rendered

My experience rst eye says this is proper formatting for a list ;) So it should be good.

**Comment by jorisvandenbossche** [comment]
Thanks @kylekeppler !

**Pull request merged**